src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health) {
+    this.name = name;
+    this.health = health || 100;
+    Animal.addAlive(this);
+  }
+
+  static addAlive(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static removeDead() {
+    return Animal.alive.filter(animal => animal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health);
+    this.hidden = hidden || false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.demage = 50;
+  }
+
+  bite(herbivore) {
+    if (
+      herbivore instanceof Herbivore && !herbivore.hidden) {
+      herbivore.health -= this.demage;
+      Animal.alive = Animal.removeDead();
```

Comment:
It's not good practise to call this method every time bite one is called. Just add check
```suggestion
      if (herbivore.health <= 0) {
        Animal.alive = Animal.removeDead();
      }
```

Reply:
Thank you for feedback!) All recommendations were applied.